src/main/java/seedu/address/logic/commands/AddNoteCommand.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_IC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NOTE;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.IdentityCardNumberMatchesPredicate;
+import seedu.address.model.person.Note;
+import seedu.address.model.person.Person;
+
+/**
+ * Changes the note of an existing person in the address book.
+ */
+public class AddNoteCommand extends Command {
+
+    public static final String COMMAND_WORD = "addnote";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the note of the person whose profile matches "
+            + "the specified IC (case-insensitive). "
+            + "Existing remark will be appended by default. To replace the original note, add -replace at "
+            + "the end of your command. E.g. addnote i/S0123456Q n/Diabetes -replace\n"
+            + "Parameters: "
+            + "[" + PREFIX_IC + "IC] "
+            + "[" + PREFIX_NOTE + "NOTE] \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_IC + " S0123456Q "
+            + PREFIX_NOTE + "Healthy.";
+
+
+    public static final String MESSAGE_ADD_NOTE_SUCCESS = "Added note to Person: %1$s";
+    public static final String MESSAGE_DELETE_NOTE_SUCCESS = "Removed note from Person: %1$s";
+    private final IdentityCardNumberMatchesPredicate ic;
+    private final Note note;
+    private final boolean isReplace;
+
+    /**
+     * @param ic of the person in the filtered person list to edit the note
+     * @param note of the person to be updated to
+     */
+    public AddNoteCommand(IdentityCardNumberMatchesPredicate ic, Note note, boolean isReplace) {
+        requireAllNonNull(ic, note);
+
+        this.ic = ic;
+        this.note = note;
+        this.isReplace = isReplace;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        model.updateFilteredPersonList(ic);
+        List<Person> lastShownList = model.getFilteredPersonList();
```

Comment:
For this, im not sure if we should just do getPersonList() instead for all the patients in the database? or do we need to filter the list with some predicate

Reply:
Maybe @tengcharmaine you could make an issue to relook at `AddNoteCommand`'s execute method for v1.3? So we don't forget. I also think it's fine to leave it for now, just for MVP.

Reply:
okay! issue created!